src/app/components/modules/gpa-calculator/gpa-calculator.component.ts

```diff
@@ -0,0 +1,118 @@
+import { MyMICDS, GetClassesResponse } from '@mymicds/sdk';
+import {
+	Component,
+	ElementRef,
+	OnDestroy,
+	OnInit,
+	ViewChild
+} from '@angular/core';
+import { SubscriptionsComponent } from '../../../common/subscriptions-component';
+
+@Component({
+	selector: 'mymicds-gpa-calculator',
+	templateUrl: './gpa-calculator.component.html',
+	styleUrls: ['./gpa-calculator.component.scss']
+})
+export class GpaCalculatorComponent extends SubscriptionsComponent
+	implements OnInit, OnDestroy {
+	@ViewChild('moduleContainer', { static: true }) moduleContainer: ElementRef;
+	displayClassesArray = ['A', 'B', 'C', 'D', 'E', 'F', 'G'];
+	classes: GetClassesResponse = null;
+	// MICDS grading scale starts at F, goes to A. Mappings retreived from guidebook
+	letterGradesArray = [
+		'N/A',
+		'A',
+		'A-',
+		'B+',
+		'B',
+		'B-',
+		'C+',
+		'C',
+		'C-',
+		'D+',
+		'D',
+		'D-',
+		'F'
+	];
+
+	calculationMappings = {
+		A: 4.0,
+		'A-': 3.67,
+		'B+': 3.33,
+		B: 3.0,
+		'B-': 2.67,
+		'C+': 2.33,
+		C: 2.0,
+		'C-': 1.67,
+		'D+': 1.33,
+		D: 1.0,
+		'D-': 0.67,
+		F: 0.0
+	};
+
+	inputGrades = ['', '', '', '', '', '', ''];
+	hasClickedCalculationButton = false;
+	calculationOutputDisplayString = 'Something\'s Broken';
+
+	constructor(private myMicds: MyMICDS) {
+		super();
+	}
+
+	ngOnInit() {
+		// create display list of classes, if there's not an alias just call it A, B etc...
+		this.myMicds.classes.get().subscribe(response => {
+			this.classes = response;
```

Comment:
From what I can tell, you're only using the `classes` property of the response and it's only being used in the observable callback, so there's no reason to assign it to a `this` property

Reply:
u rite

Reply:
When you make the change, don't forget that we can use parameter destructuring in the callback